src/docbuilder/metadata.rs

```diff
@@ -192,6 +198,18 @@ impl Metadata {
     }
 }
 
+fn fetch_manifest_tables<'a>(manifest: &'a Value) -> Option<&'a Map<String, Value>> {
```

Comment:
What's the reason for a separate function?

Reply:
Makes things much cleaner because of how chained the internals are

Reply:
I'd rather not separate out functions unless it represents another level of abstraction (e.g. we expect the implementation to change) or the code is used more than once. Can you make this a temporary variable in `from_str` instead? That will make it easier to see what's going on without jumping around in the file.

Reply:
I moved the function up to where it was used, as the entire reason I made it a function (That I totally forgot about) is that the parent function doesn't return an option, but the internals of `fetch_manifest_tables` uses *tons*, and the separate function allows using the try operator for much cleaner code, since we ultimately want to handle the `None` variant. I attempted to make it a closure, but the lifetimes didn't want to play nice